LAUNCH.md

```diff
@@ -0,0 +1,200 @@
+venv          : None
+verbose       : False
+```
+
+## 🥳 User guide
+
+In a word, use `launch.py` to fill in an sbatch template and submit either
+a single job from the command-line, or a list of jobs from a `yaml` file.
+
+Examples:
+
+```sh
+# using default job configuration, with script args from the command-line:
+$ python launch.py user=$USER logger.do.online=False
+
+# overriding the default job configuration and adding script args:
+$ python launch.py --template=sbatch/template-venv.sh \
+    --venv='~/.venvs/gfn' \
+    --modules='python/3.7 cuda/11.3' \
+    user=$USER logger.do.online=False
+
+# using a yaml file to specify multiple jobs to run:
+$ python launch.py --jobs=jobs/comp-sg-lp/v0" --mem=32G
+```
+
+### 🤓 How it works
+
+1. All experiment files should be in `external/jobs`
+    1. Note that all the content in `external/` is **ignored by git**
+2. You can nest experiment files infinitely, let's say you work on crystals and call your experiment `explore-losses.yaml` then you could put your config in `external/jobs/crystals/explore-losses.yaml`
+3. An experiment file contains 2 main sections:
+    1. `shared:` contains the configuration that will be, you guessed it, shared across jobs.
+    2. `jobs:` lists configurations for the SLURM jobs that you want to run. The `shared` configuration will be loaded first, then updated from the `run`'s.
+4. Both `shared` and `job` dicts contain (optional) sub-sections:
+    1. `slurm:` contains what's necessary to parameterize the SLURM job
+    2. `script:` contains a dict version of the command-line args to give `main.py`
+
+    ```yaml
+    script:
+      gflownet:
+        optimizer:
+          lr: 0.001
+
+    # is equivalent to
+    script:
+      gflownet.optimizer.lr: 0.001
+
+    # and will be translated to
+    python main.py gflownet.optimizer.lr=0.001
+    ```
+
+5. Launch the SLURM jobs with `python launch.py --jobs=crystals/explore-losses`
+    1. `launch.py` knows to look in `external/jobs/` and add `.yaml`
+    2. You can overwrite anything from the command-line: the command-line arguments have the final say and will overwrite all the jobs' final dicts. Run `python launch.py -h` to see all the known args.
+    3. You can also override `script` params from the command-line: unknown arguments will be given as-is to `main.py`. For instance `python launch.py --jobs=crystals/explore-losses --mem=32G env.some_param=value` is valid
+6. `launch.py` loads a template (`sbatch/template-conda.sh`) by default, and fills it with the arguments specified, then writes the filled template in `external/launched_sbatch_scripts/crystals/` with the current datetime and experiment file name.
+7. `launch.py` executes `sbatch` in a subprocess to execute the filled template above
+8. A summary yaml is also created there, with the exact experiment file and appended `SLURM_JOB_ID`s returned by `sbatch`
+
+### 📝 Case-study
```

Comment:
Would it make sense to add the example file in the case study to the repo?

Reply:
Yeah, I'll try to find a place for it that does not confuse ppl. It's just that it's supposed to sit in `external/` which is ignored 😅

Reply:
It's now in `sbatch/`